### PR TITLE
Mellanox MSN2700, MSN2100 and MSN2410 platforms ONL support.

### PR DESCRIPTION
**Pull request by micshy**
Signed-off-by: Michael Shych <michaelsh@mellanox.com>

### CODE REVIEW

**Comment by micshy** [comment]
Please review new pull request of ONL support of 3 Mellanox platforms: MSN2700, MSN2100 and MSN2410.
Thanks, Michael.

File: packages/platforms/mellanox/x86-64/x86-64-mlnx-msn2100/onlp/builds/src/module/src/ledi.c [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+    int  len, local_id = 0;
+    uint8_t data[driver_value_len] = {0};
+    char fullpath[50] = {0};
+
+    VALIDATE(id);
+
+    local_id = ONLP_OID_ID_GET(id);
+
+    /* get fullpath */
+    snprintf(fullpath, sizeof(fullpath), "%s%s", prefix_path, file_names[local_id]);
```
Comment (jnealtowns) [inline packages/platforms/mellanox/x86-64/x86-64-mlnx-msn2100/onlp/builds/src/module/src/ledi.c]: This pattern occurs so frequently that the `onlp_file_read()` function already handles it for you:
https://github.com/opencomputeproject/OpenNetworkLinux/blob/f77e29cd513b8cede557af820750d83c57985d00/packages/base/any/onlp/src/onlplib/module/inc/onlplib/file.h#L48

You don't need the `fullpath` temporary variable. You can call `onlp_file_read` as follows:

`onlp_file_read(data, sizeof(data), &len, "%s%s", prefix_path, file_names[local_id]`

File: packages/platforms/mellanox/x86-64/x86-64-mlnx-msn2100/onlp/builds/src/module/src/psui.c [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+psu_power_info_get(int id, char *node, int *value)
+{
+    int len, ret = 0;
+    char buf[PSU_NODE_MAX_INT_LEN + 1] = {0};
+    char node_path[PSU_NODE_MAX_PATH_LEN] = {0};
+
+    *value = 0;
+
+    sprintf(node_path, PSU_POWER_PREFIX, id, node);
+    ret = onlp_file_read((uint8_t*)buf, sizeof(buf), &len, node_path);
```
Comment (jnealtowns) [inline packages/platforms/mellanox/x86-64/x86-64-mlnx-msn2100/onlp/builds/src/module/src/psui.c]: node_path

File: packages/platforms/mellanox/x86-64/x86-64-mlnx-msn2100/onlp/builds/src/module/src/thermali.c [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+    local_id = ONLP_OID_ID_GET(id);
+
+    /* Set the onlp_oid_hdr_t and capabilities */
+    *info = linfo[local_id];
+
+    /* get fullpath */
+    snprintf(fullpath, sizeof(fullpath), "%s/%s", prefix_path, last_path[local_id]);
+
+    rv = onlp_file_read((uint8_t*)r_data, sizeof(r_data), &len, fullpath);
```
Comment (jnealtowns) [inline packages/platforms/mellanox/x86-64/x86-64-mlnx-msn2100/onlp/builds/src/module/src/thermali.c]: fullpath

File: packages/platforms/mellanox/x86-64/x86-64-mlnx-msn2410/onlp/builds/src/module/src/ledi.c [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+    int  len, local_id = 0;
+    uint8_t data[driver_value_len] = {0};
+    char fullpath[50] = {0};
+
+    VALIDATE(id);
+
+    local_id = ONLP_OID_ID_GET(id);
+
+    /* get fullpath */
+    snprintf(fullpath, sizeof(fullpath), "%s%s", prefix_path, file_names[local_id]);
```
Comment (jnealtowns) [inline packages/platforms/mellanox/x86-64/x86-64-mlnx-msn2410/onlp/builds/src/module/src/ledi.c]: You get the idea :-)

**Comment by jnealtowns** [comment]
@micshy I'm going to merge your changes. Thank you for your contribution!
I will submit a change to you for review which demonstrates how to use the (new) common infrastructure for gathering the contents of the system eeprom. This will only affect your ONIE data export code but will make it simpler for you to implement.

**Comment by micshy** [comment]
Hi Jeff,

Thank you for review and new infrastructure API.
I’ll check it when it will be ready.

Regards,
    Michael.

**Pull request merged**